bytecomp/bytelink.ml

```diff
@@ -371,42 +365,43 @@ let link_bytecode ?final_name tolink exec_name standalone =
        end;
        let output_fun = output_bytes outchan
        and currpos_fun () = pos_out outchan - start_code in
-       List.iter (link_file output_fun currpos_fun) tolink;
+       List.iter (link_file sym output_fun currpos_fun) tolink;
        if check_dlls then Dll.close_all_dlls();
        (* The final STOP instruction *)
        output_byte outchan Opcodes.opSTOP;
        output_byte outchan 0; output_byte outchan 0; output_byte outchan 0;
-       Bytesections.record outchan "CODE";
+       Bytesections.record toc_writer "CODE";
        (* DLL stuff *)
        if standalone then begin
          (* The extra search path for DLLs *)
-         output_stringlist outchan !Clflags.dllpaths;
-         Bytesections.record outchan "DLPT";
+         output_string outchan (concat_null_separated !Clflags.dllpaths);
+         Bytesections.record toc_writer "DLPT";
```

Comment:
Another possible improvement: strings like `"DLPT"`, which are referred to from multiple places (`Symtable`, `Bytelink`) could be put in a section name module so they are only in one place

Reply:
This has been done in #11623